tests/bsim/bluetooth/host/iso/bis/src/bis_receiver.c

```diff
@@ -74,10 +81,24 @@ static void iso_log_data(uint8_t *data, size_t data_len)
 static void iso_recv(struct bt_iso_chan *chan, const struct bt_iso_recv_info *info,
 		     struct net_buf *buf)
 {
+	static size_t rx_cnt;
+
+	if (IS_FLAG_SET(flag_data_received)) {
+		return;
+	}
+
 	if (info->flags & BT_ISO_FLAGS_VALID) {
 		LOG_DBG("Incoming data channel %p len %u", chan, buf->len);
 		iso_log_data(buf->data, buf->len);
-		SET_FLAG(flag_data_received);
+
+		if (memcmp(buf->data, mock_iso_data, buf->len) != 0) {
+			TEST_FAIL("Unexpected data received");
+		} else if (rx_cnt++ > RX_CNT_TO_PASS) {
+			LOG_INF("Data received");
+			SET_FLAG(flag_data_received);
+		}
+	} else {
+		TEST_FAIL("Failed to receive");
```

Comment:
This new validation may need this fix: https://github.com/zephyrproject-rtos/zephyr/pull/80788/commits/736bf5b5c3f961341a122effdad699af24832b5d

Reply:
Please put it back, it will be a good coverage when https://github.com/zephyrproject-rtos/zephyr/pull/80788 is merged.